File: TheaterSuggestions/Java/ExternalApis/AuditoriumLayoutApi/src/main/java/com/baasie/AuditoriumLayoutApi/controllers/AuditoriumSeatingController.java
package com.baasie.AuditoriumLayoutApi.controllers;


import com.baasie.ExternalDependencies.IProvideAuditoriumLayouts;
import com.baasie.ExternalDependencies.auditoriumlayoutrepository.AuditoriumDto;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("api/data_for_auditoriumSeating")
public class AuditoriumSeatingController {

    private IProvideAuditoriumLayouts provideAuditoriumLayouts;

    public AuditoriumSeatingController(IProvideAuditoriumLayouts provideAuditoriumLayouts) {
        this.provideAuditoriumLayouts = provideAuditoriumLayouts;
    }

    // GET api/data_for_auditoriumSeating/5
    @GetMapping(value = "/{showId}", produces = "application/json")
    public AuditoriumDto get(@PathVariable String showId) {
        return provideAuditoriumLayouts.getAuditoriumSeatingFor(showId);
    }
}

File: TheaterSuggestions/CSharp/AuditoriumLayout.Api/Controllers/AuditoriumSeatingController.cs
using System.Threading.Tasks;
using ExternalDependencies;
using Microsoft.AspNetCore.Mvc;

namespace AuditoriumLayout.Api.Controllers
{
    [ApiController]
    [ApiVersion("1.0")]
    [Route("api/v{version:apiVersion}/data_for_auditoriumSeating")]
    public class AuditoriumSeatingController : ControllerBase
    {
        private readonly IProvideAuditoriumLayouts _provideAuditoriumLayouts;

        public AuditoriumSeatingController(IProvideAuditoriumLayouts provideAuditoriumLayouts)
        {
            _provideAuditoriumLayouts = provideAuditoriumLayouts;
        }

        // GET api/data_for_auditoriumSeating/5
        [HttpGet("{showId}")]
        public async Task<ActionResult<AuditoriumDto>> Get(string showId)
        {
            return await _provideAuditoriumLayouts.GetAuditoriumSeatingFor(showId);
        }
    }
}